test/Table_Tests/src/Database/Postgres_Spec.enso

```diff
@@ -338,6 +341,96 @@ postgres_specific_spec connection db_name setup =
             w4.requested_type . should_equal (Value_Type.Decimal precision=Nothing scale=Nothing)
             w4.actual_type . should_equal Value_Type.Float
 
+        Test.specify "should round-trip timestamptz column, preserving intsant but converting to UTC" <|
+            table_name = Name_Generator.random_name "TimestampTZ"
+            table = connection.create_table table_name [Column_Description.Value "A" (Value_Type.Date_Time with_timezone=True)] primary_key=[]
+
+            dt1 = Date_Time.new 2022 05 04 15 30
+            dt2 = Date_Time.new 2022 05 04 15 30 zone=(Time_Zone.utc)
+            dt3 = Date_Time.new 2022 05 04 15 30 zone=(Time_Zone.parse "US/Hawaii")
+            dt4 = Date_Time.new 2022 05 04 15 30 zone=(Time_Zone.parse "Europe/Warsaw")
+            v = [dt1, dt2, dt3, dt4]
+
+            Problems.assume_no_problems <| table.update_rows (Table.new [["A", v]]) update_action=Update_Action.Insert
+
+            tz_zero = Time_Zone.parse "Z"
+            v_at_z = v.map dt-> dt.at_zone tz_zero
+            table.at "A" . to_vector . should_equal v_at_z
+            table.at "A" . to_vector . should_equal_tz_agnostic v
+
+            ## We also check how the timestamp column behaves with interpolations:
+               (See analogous test showing a less nice behaviour without timezones below.)
+            v.each my_dt-> Test.with_clue my_dt.to_text+" " <|
+                # Depending on test runner's timezone, the `my_dt` may be equal to 1 or 2 entries in `v`.
+                equals_tz_agnostic dt1 dt2 =
+                    dt1.at_zone tz_zero == dt2.at_zone tz_zero
+                local_equals = v.filter (equals_tz_agnostic my_dt)
+                [1, 2].should_contain local_equals.length
+
+                Test.with_clue " == "+local_equals.to_text+": " <|
+                    t2 = table.filter "A" (Filter_Condition.Equal to=my_dt)
+                    t2.row_count . should_equal local_equals.length
+                    t2.at "A" . to_vector . should_equal_tz_agnostic local_equals
+
+        Test.specify "will round-trip timestamp column without timezone by converting it to UTC" <|
+            table_name = Name_Generator.random_name "Timestamp"
+            table = connection.create_table table_name [Column_Description.Value "A" (Value_Type.Date_Time with_timezone=False)] primary_key=[]
+            Problems.assume_no_problems table
+
+            dt1 = Date_Time.new 2022 05 04 15 30
+            dt2 = Date_Time.new 2022 05 04 15 30 zone=(Time_Zone.utc)
+            dt3 = Date_Time.new 2022 05 04 15 30 zone=(Time_Zone.parse "US/Hawaii")
+            dt4 = Date_Time.new 2022 05 04 15 30 zone=(Time_Zone.parse "Europe/Warsaw")
+            v = [dt1, dt2, dt3, dt4]
+
+            w = Problems.expect_only_warning Inexact_Type_Coercion <|
+                table.update_rows (Table.new [["A", v]]) update_action=Update_Action.Insert
+            w.requested_type . should_equal (Value_Type.Date_Time with_timezone=True)
```

Comment:
Why would the requested type be `with_timezone=True` here?

Reply:
Okay maybe I will split it into separate lines to make it clearer.
```
src_table = Table.new [["A", v]]
src_table.at "A" . value_type . should_equal (Value_Type.Date_Time with_timezone=True)
table.update_rows src_table update_action=Update_Action.Insert
```

I hope that makes it clearer.

The _source_ table requests to upload Date_Time with timezone, but the target's table actual type is without timezone.